projects/packages/waf/src/class-waf-runner.php

```diff
@@ -366,24 +386,39 @@ public static function deactivate() {
 
 		// Empty the rules entrypoint file.
 		if ( ! $wp_filesystem->put_contents( self::get_waf_file_path( Waf_Rules_Manager::RULES_ENTRYPOINT_FILE ), "<?php\n" ) ) {
-			throw new \Exception( 'Failed to empty rules.php file.' );
+			throw new File_System_Exception( 'Failed to empty rules.php file.' );
 		}
 	}
 
 	/**
 	 * Handle updates to the WAF
+	 *
+	 * @throws Waf_Exception If generating the standalone bootstrap file fails.
+	 * @throws Waf_Exception If generating the rule files fails.
+	 *
+	 * @return void
 	 */
 	public static function update_waf() {
-		Waf_Rules_Manager::update_rules_if_changed();
+		$rules_generated_exception = null;
+
+		try {
+			Waf_Rules_Manager::update_rules_if_changed();
+		} catch ( Waf_Exception $e ) {
+			$rules_generated_exception = $e;
+		}
+
 		// Re-generate the standalone bootstrap file on every update
 		// TODO: We may consider only doing this when the WAF version changes
 		try {
 			( new Waf_Standalone_Bootstrap() )->generate();
-		} catch ( \Exception $e ) {
-			return new WP_Error( 'waf_update_failed', $e->getMessage() );
-		}
 
-		return true;
+			// Throw the exception from generating the rules after we've generated the standalone bootstrap file.
+			if ( $rules_generated_exception ) {
+				throw $rules_generated_exception;
+			}
+		} catch ( Waf_Exception $e ) {
+			throw $e;
+		}
```

Comment:
Can also be removed 🙂

Reply:
Resolved via https://github.com/Automattic/jetpack/commit/c1d3effd93c8fd23355781bca6b324383c18d498